tests/packages/test_main.py

```diff
@@ -254,3 +254,12 @@ def test_dependency_from_pep_508_with_python_full_version_pep440_compatible_rele
     assert dep.name == "pathlib2"
     assert str(dep.constraint) == "*"
     assert dep.python_versions == "~=3.4 || <3"
+
+
+def test_dependency_from_pep_508_with_invalid_python_version_constraint():
+    name = 'pytest-mypy ; python_version < "3.10" and python_version > "3"'
+    dep = Dependency.create_from_pep_508(name)
+
+    assert dep.name == "pytest-mypy"
+    assert str(dep.constraint) == "*"
+    assert dep.python_versions == "<3.10 >=4"
```

Comment:
Based on https://www.python.org/dev/peps/pep-0508/#versions and https://www.python.org/dev/peps/pep-0508/#environment-markers, I'd say Poetry should not consider `> "3"` to be equivalent to `>= "4". These versions should be compared using PEP440, and with PEP440 `3.6` is greater than `3`.

```python
>>> import packaging
>>> from packaging import version
>>> version.parse("3.6") > version.parse("3")
True
>>> version.parse("3.6") >= version.parse("4")
False
```